src/App.js

```diff
@@ -1,26 +1,120 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import Header from './components/Header';
+import Footer from './components/Footer';
+import Main from './components/Main';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: '',
+      items: [],
+      selectedList: 0,
+      checked: false,
+    };
+
+    this.addToValue = (event) => {
+      this.setState({
+        value: event.target.value,
+      });
+    };
+
+    this.addToItems = () => {
+      if (this.state.value) {
+        this.setState(prevState => ({
+          items: [
+            ...prevState.items,
+            {
+              value: this.state.value,
+              done: false,
+            },
+          ],
+
+          value: '',
+        }));
+      }
+    };
+
+    this.handleClickToDone = (value) => {
+      for (const key of this.state.items) {
+        if (key.value === value) {
+          key.done = !key.done;
+        }
+      }
+
+      this.setState({
+        items: this.state.items,
+
+      });
+    };
+
+    this.showAllItems = () => {
+      this.setState({ selectedList: 0 });
+    };
+
+    this.showActiveItems = () => {
+      this.setState({ selectedList: 1 });
+    };
+
+    this.showCompletedItems = () => {
+      this.setState({ selectedList: 2 });
+    };
+
+    this.removeAllCompletedItems = () => {
+      const items = this.state.items.filter(item => !item.done);
+      this.setState({ items });
+    };
+
+    this.removeCompletedItem = (value) => {
+      const newValue = this.state.items.filter(item => item.value !== value);
+
+      this.setState({ items: newValue });
+    };
+
+    this.checkedAllItems = () => {
+      for (const key of this.state.items) {
+        key.done = !this.state.checked;
+      }
+
+      this.state.checked = !this.state.checked;
+      this.setState({
+        items: this.state.items,
+        checked: this.state.checked,
+      });
+    };
+  }
+
+  render() {
+    return (
+      <section className="todoapp">
+        <Header
+          value={this.state.value}
+          handleChange={this.addToValue}
+          handleSubmit={this.addToItems}
+        />
+        <Main
+          items={this.state.selectedList === 0
+            ? this.state.items
+            : this.state.selectedList === 1
+              ? this.state.items.filter(item => !item.done)
+              : this.state.items.filter(item => item.done)}
```

Comment:
Avoid nested ternary. It's quite hard to understand what does this piece of code do.
Suggestion: create method `filterItems` which returns an array of filtered todos.
```jsx
filterItems() {
  // we've already renamed `selectedList` to `filterBy`
  switch (this.state.filterBy) {
    case FILTER.all: {
      return this.state.items;
    }
    case FILTER.active: {
      return this.state.items.filter((item) => !item.done);
    }
    case FILTER.completed: {
      return this.state.items.filter((item) => item.done);
    }
    default: {
      return this.state.items;
    }
  }
}

// ...
// later, in render

<Main
  items={this.filterItems}
  // ...
```
It's good point to avoid render and calculation logic